### PR TITLE
Remove dead jobs from ci.ros2.org

### PR DESCRIPTION
**Pull request by clalancette**
Neither the armhf jobs nor the turtlebot jobs are currently viable.  Remove both of them in this pull request.

I've verified that this doesn't actually change any job configurations on https://ci.ros2.org by running `create_jenkins_jobs.py` in dry-run mode.  Once this is approved and merged, I'll manually delete the armhf and turtlebot jobs from the farm, and then they will not be created again.

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
I'm still using CI to produce armhf archives for Dashing ([patch release 7](https://github.com/ros2/ros2/releases/tag/release-dashing-20200722)). I'd prefer not to remove armhf-support from the internals and the manual jobs. I recall discussing this change offline but I must have misinterpreted the full extend of the proposed change.

**Comment by clalancette** [comment]
> I'm still using CI to produce armhf archives for Dashing ([patch release 7](https://github.com/ros2/ros2/releases/tag/release-dashing-20200722)). I'd prefer not to remove armhf support from the internals and the manual jobs. I recall discussing this change offline but I must have misinterpreted the full extend of the proposed change.

Oh,I forgot we were still supporting it for Dashing.  Good call.

In that case, I'll revert most of that and just remove the nightly armhf jobs.  That would leave all other armhf jobs available.  Does that satisfy your needs?

**Comment by nuclearsandwich** [comment]
> In that case, I'll revert most of that and just remove the nightly armhf jobs. That would leave all other armhf jobs available. Does that satisfy your needs?

Yes I just need the manual  ci_ (which will preserve the test_ci_ job as well) and the manual packaging job but not the nightlies or nightly packaging job.

**Comment by clalancette** [comment]
All right, I've rewritten it to just get rid of the nightly jobs.  The following ones will still be available:

```
Skipped 'ci_linux-armhf' because the config is the same (dry run)
Skipped 'test_ci_linux-armhf' because the config is the same (dry run)
Skipped 'ci_packaging_linux-armhf' because the config is the same (dry run)
Skipped 'packaging_linux-armhf' because the config is the same (dry run)
```

File: create_jenkins_job.py [hunk 1]
```diff
@@ -264,15 +263,16 @@ def create_job(os_name, job_name, template_file, additional_dict):
...
-            'time_trigger_spec': PERIODIC_JOB_SPEC,
-            'mailer_recipients': DEFAULT_MAIL_RECIPIENTS,
-        })
+        if os_name != 'linux-armhf':
+            job_name = 'nightly_' + job_os_name + '_debug'
+            if os_name == 'windows':
+                job_name = job_name[:15]
+            create_job(os_name, job_name, 'ci_job.xml.em', {
+                'cmake_build_type': 'Debug',
+                'disabled': False,
```
Comment (dirk-thomas) [inline create_jenkins_job.py:272]: I would guess this line isn't necessary anymore.

Same below.

Comment (clalancette) [reply create_jenkins_job.py]: Good point.  Removed in ea4fd7c.

**Comment by clalancette** [comment]
Thanks, merging.  I'll deploy next (though that is mostly a no-op).  Then I'll manually delete the dead jobs.

**Pull request merged**